posts/models.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+from django.conf import settings
+
+
+class Tag(models.Model):
+    name = models.CharField(max_length=50, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Post(models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name='posts')
+    text_content = models.TextField()
+    image = models.ImageField(upload_to='post_images/', blank=True, null=True)
+    tags = models.ManyToManyField(Tag, related_name='posts')

```

Comment:
Once my post is deleted, the tag value for the previous tag should also be deleted.